proto/cosmos/bank/v1beta1/query.proto

```diff
@@ -83,6 +83,12 @@ service Query {
     option (google.api.http).get               = "/cosmos/bank/v1beta1/denoms_metadata/{denom}";
   }
 
+  // DenomsMetadata queries the client metadata of a given coin denomination.
+  rpc DenomMetadataByQueryString(QueryDenomMetadataByQueryStringRequest)
+      returns (QueryDenomMetadataByQueryStringResponse) {
+    option (cosmos.query.v1.module_query_safe) = true;
+    option (google.api.http).get               = "/cosmos/bank/v1beta1/denoms_metadata_by_query_string";
```

Comment:
```suggestion
    option (google.api.http).get               = "/cosmos/bank/v1beta1/denoms_metadata_by_query";
```

Reply:
IMO, query string seems like the correct term to use, is there a reason for suggestion above?

Reply:
Yes, a few items:

* It makes an already verbose query even more verbose without adding context
* The fact that denom is a string is already available based on the proto schema
* Generally, you dont wan't to annotate a query's path (e.g. /users/by_name vs /users/by_name_string), the latter doesn't give you anything as it's superfluous IMO

Reply:
Oh I was referring to query string as a single term here though (https://en.wikipedia.org/wiki/Query_string)

Reply:
Not a blocker for me, just my two cents :p

we can merge